File: InsertionSort.py
##Insertion Sort in Python

import random
import timeit

def Insertion_Sort(x):
    i = 1
    min = 0
    for i in range(len(x)):
        ##Find the index where min element is present
        for j in range(i, len(x)):
            if x[j] < x[min]:
                min = j
        
        ##swap the element with first element
        temp = x[min]
        x[min] = x[i - 1]
        x[i - 1] = temp

        min = i
    
    print("Sorted Array : ", x)
    return


size = int(input("Enter Length of Array : "))
N = int(input("Number of Execution : "))
t = 0.0

for i in range(N):
    X = []
    ##Random Integeres Input
    for j in range(0, size):
        X.append(random.randint(-size * 2, size * 2))
    print("Unsorted Array : ", X)
    t += timeit.timeit(lambda : Insertion_Sort(X), number = 1)

print("Average Execution Time for Insertion sort : ", float(t / N))

File: Insertion_Sort.cpp
// Implementation of Insertion Sort

#include <iostream>
using namespace std;

void insertion_Sort(int A[], int length) {
    int i, max, j = length - 1;

    while (j > 0) {
        max = 0;
        for (i = 0; i <= j; i++) {
            if (A[max] < A[i]) {
                max = i;
            }
        }

        // Swap Values
        int Temp = A[max];
        A[max] = A[j];
        A[j] = Temp;
        j --;
    }

    // Print Array
    cout<<"\nSorted Array :";
    for (int i = 0; i < length; i++) {
        cout << " " << A[i];
    }

    return;
}



int main(void){
    int length = 5;
    int A[length] = {10, 4, 7, -9, 2};

    cout<<"Unsorted Array :";
    for(int i = 0; i < length; i ++) {
        cout<< " " << A[i];
    }

    // Call Insertion Sort Function
    insertion_Sort(A, length);

    return 0;
}